textdb/textdb-dataflow/src/main/java/edu/uci/ics/textdb/dataflow/neextrator/NamedEntityExtractor.java

```diff
@@ -72,7 +85,168 @@ public void open() throws Exception {
+            }
+            List<Span> spanList = new ArrayList<>();
+            for (Attribute attribute : searchInAttributes) {
+                String fieldName = attribute.getFieldName();
+                IField field = sourceTuple.getField(fieldName);
+                spanList.addAll(extractNESpans(field, fieldName));
+            }
+
+            ITuple returnTuple = Utils.getSpanTuple(sourceTuple.getFields(), spanList, returnSchema);
+            sourceTuple = sourceOperator.getNextTuple();
+            return returnTuple;
+        }
+    }
+
+    /**
+     * @param iField
+     * @return a List of spans of the extracted information
+     * @about This function takes an (TextField) IField and a String
+     * (the field's name) as input and uses the Stanford NLP package to process the field.
+     * It returns a list of spans
+     * In the returning span: Value -> the word itself
+     * Key   -> NE_Constant
+     * @overview Using the Stanford NLP package to process the textField value.
+     * First set up a pipeline of Annotators for the string: TokenizerAnnotator,
+     * SentencesAnnotator, PartOfSpeechAnnotator, LemmaAnnotator and
+     * NamedEntityTagAnnotator. The order is mandatory because they have
+     * dependencies. After the pipeline, each token is wrapped as a CoreLabel
+     * and each sentence is wrapped as CoreMap. Each annotator adds its
+     * annotation to the CoreMap(sentence) or CoreLabel(token) object.
+     * <p>
+     * After the pipeline is done, scan each CoreLabel(token) for
+     * its NamedEntityAnnotation, if it's a valid value (not 'O'),
+     * then makes it a span and add to the return list.
+     * The Stanford NLP constants are mapped into the NE constants.
+     * The NLP package has annotations for the start and end position of a token
+     * and it perfectly matches the span design so we just use them.
+     *
+     */
+    private List<Span> extractNESpans(IField iField, String fieldName) {
+        List<Span> spanList = new ArrayList<>();
+        String text = (String) iField.getValue();
+        Properties props = new Properties();
+        props.setProperty("annotators", "tokenize, ssplit, pos, lemma, ner");
+        StanfordCoreNLP pipeline = new StanfordCoreNLP(props);
+        Annotation documentAnnotation = new Annotation(text);
+        pipeline.annotate(documentAnnotation);
+        List<CoreMap> sentences = documentAnnotation.get(CoreAnnotations.SentencesAnnotation.class);
+        for (CoreMap sentence : sentences) {
+            for (CoreLabel token : sentence.get(CoreAnnotations.TokensAnnotation.class)) {
+                String NLPConstant = token.get(CoreAnnotations.NamedEntityTagAnnotation.class);
+                if (!NLPConstant.equals("O")) {
+
+                    String NEConstant = getNEConstant(NLPConstant);
+                    int start = token.get(CoreAnnotations.CharacterOffsetBeginAnnotation.class);
```

Comment:
I want to confirm if we are using "gap offsets". Check http://www.ibm.com/support/knowledgecenter/SSPT3X_4.0.0/com.ibm.swg.im.infosphere.biginsights.aqlref.doc/doc/data-model.html

For the text `Amelia Earhart is a pilot.`, they have 

```
  0   1   2   3   4  5  6
    A   m   e   l   i   a
```

We should make sure we use gap offsets for start and end.